integration/pull_modes_test.go

```diff
@@ -252,3 +253,93 @@ func testV1IsNotUsedWhenDisabled(t *testing.T, imgName string) {
 		t.Fatalf("expected no digest, got %s", indexDigestUsed)
 	}
 }
+
+func TestDanglingV2Annotation(t *testing.T) {
```

Comment:
Not a requirement but I've started documenting for some of the lengthier tests the behavior I am working to preserve. The PR description is a pretty good writeup IMO. 

In a perfect world, it should be straightforward to read it from the test itself, but our test framework makes that a little difficult because we are doing some pretty complex stuff.